From 0442139f42c16bcd9777abb0def77e05fb743596 Mon Sep 17 00:00:00 2001
From: Fernando Reyna <fernando.m.reyna.diaz@intel.com>
Date: Tue, 23 Jun 2015 10:43:09 -0500
Subject: [PATCH] atomisp2: fix kernel panic when VIDIOC_S_FMT is set to 0

Kernel Panic occurs from user space when the VIDIOC_S_FMT
is set to 0 and when streamon is performing the DUT is
immediatelyy reboots.

Change-Id: I770c1590db1038a5cb400b0b29c938367cbe766a
Original-Tracked-on: https://jira01.devtools.intel.com/browse/GMINL-11122
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7449
Signed-off-by: Fernando Reyna <fernando.m.reyna.diaz@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/61811
Build: IMIN (SC)
Reviewed-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Reviewed-by: Qingwu Zhang <qingwu.zhang@intel.com>
Approver: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 875d5a3..3538251 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4658,6 +4658,9 @@ int atomisp_try_fmt(struct video_device *vdev, struct v4l2_format *f,
 		fmt = atomisp_output_fmts;
 	}
 
+	if (f->fmt.pix.width <= 0 || f->fmt.pix.height <= 0)
+		return -EINVAL;
+
 	snr_mbus_fmt.code = fmt->mbus_code;
 	snr_mbus_fmt.width = f->fmt.pix.width;
 	snr_mbus_fmt.height = f->fmt.pix.height;
-- 
1.9.1

